File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.go
func firstUniqChar(s string) int {
    var chars [26]int

    for _,c := range s{
        chars[c-'a']++
    }
    for i,c := range s {
        if chars[c-'a'] == 1{
            return i
        }
    }
    return -1
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        res = -1
        al = dict()
        for i in s:
            if i not in al:
                al[i] = 1
            else:
                al[i] = al[i]+1
        for i in range(len(s)):
            if al[s[i]]== 1:
                res = i
                break
        return res
